tower_cli/api.py

```diff
@@ -18,16 +18,69 @@
 import functools
 import json
 import warnings
+from datetime import datetime as dt
 
 from requests.exceptions import ConnectionError, SSLError
 from requests.sessions import Session
 from requests.models import Response
 from requests.packages import urllib3
+from requests.auth import AuthBase
 
 from tower_cli.conf import settings
 from tower_cli.utils import data_structures, debug, secho, exceptions as exc
 
 
+TOWER_DATETIME_FMT = r'%Y-%m-%dT%H:%M:%S.%fZ'
+
+
+class TowerTokenAuth(AuthBase):
+
+    def __init__(self, username, password, cli_client):
+        self.username = username
+        self.password = password
+        self.cli_client = cli_client
+
+    def _acquire_token(self):
+        return self.cli_client._make_request(
+            'POST',
+            self.cli_client.prefix + 'authtoken/',
+            [], {
+                'data': json.dumps({'username': self.username,
+                                    'password': self.password}),
+                'headers': {
+                    'Content-Type': 'application/json'
+                }
+            }
+        ).json()
+
+    def _get_auth_token(self):
+        try:
+            with open('/tmp/tower_cli_token.json') as f:
+                token_json = json.load(f)
```

Comment:
I think one requirement for this will be that this file isn't world-readable. For that, I think it would be good to have other eyes on it aside from mine.

Aside from that detail, I'm very happy with the your implementation here, managing the expiration and falling back to create a new one.